src/main.rs

```diff
@@ -143,6 +178,11 @@ async fn run_queue(cli: &Config) -> Result<(), openssh::Error> {
     let mut job_queue = JobQueue::new();
     let mut host_index;
     loop {
+        // Check cancel.
+        if *cancelled.lock().await {
```

Comment:
Using `tokio::sync::Notify` for `cancelled` would remove the need to busy loop here.